com.unity.render-pipelines.core/Editor/Utilities/MenuUtilities.cs

```diff
@@ -0,0 +1,169 @@
+            => s_AddMenuItem(element.attribute.path,
+                string.Empty,
+                false,
+                element.attribute.priority,
+                element.action,
+                element.validate);
+
+        static void UnregisterMenu(MenuItemData element)
+            => s_RemoveMenuItem(element.attribute.path);
+
+        [InitializeOnLoadMethod]
+        static void ConstructMenuItemForRenderPipeline()
+        {
+            // Construct the MenuMap
+            // Possible scenario to take into acount:
+            //   - validate method can come before action method
+            //   - all packages add with same attribute (renderPipelineAssetTypes = { HDRPAsset, URPAsset })
+            //   - each package add its own attribute (one with renderPipelineAssetTypes = { HDRPAsset }, other with renderPipelineAssetTypes = { URPAsset })
+            var filteredMethods = TypeCache.GetMethodsWithAttribute<MenuItemForRenderPipeline>();
+            foreach (var methodInfo in filteredMethods)
+            {
+                var attribute = methodInfo.GetCustomAttribute<MenuItemForRenderPipeline>();
+                foreach (Type rpAssetType in attribute.renderPipelineAssetTypes)
+                {
+                    if (!s_MenuMap.TryGetValue(rpAssetType, out var menuItemsForRP))
+                    {
+                        menuItemsForRP = new List<MenuItemData>();
+                        s_MenuMap[rpAssetType] = menuItemsForRP;
+                    }
+
+                    var query = menuItemsForRP.Where(mi => mi.attribute.path == attribute.path);
+                    bool alreadyRegistered = query.Count() > 0;
+                    var menuItemData = alreadyRegistered ? query.First() : new MenuItemData();
+
+                    menuItemData.attribute = attribute;
+                    if (attribute.validate)
+                        menuItemData.validate = (Func<bool>)Delegate.CreateDelegate(typeof(Func<bool>), methodInfo);
+                    else
+                        menuItemData.action = (Action)Delegate.CreateDelegate(typeof(Action), methodInfo);
+
+                    if (!alreadyRegistered)
+                        menuItemsForRP.Add(menuItemData);
+                }
+            }
+
+            //activating menuitems
+            UpdateMenuItemForRenderPipeline();
+        }
+
+        static void UpdateMenuItemForRenderPipeline()
+        {
+            // remove
+            foreach (var menuItemData in s_MenuMap.Values.SelectMany(mid => mid))
+                UnregisterMenu(menuItemData); //do nothing if not registered before
+
+            // add if required
+
+            // Note: GraphicsSettings.currentRenderPipeline == null possible and should mean built-in
+            // though we cannot use null as key so it was early replaced by typeof(RenderPipelineAsset)
+            Type currentSRPType = GraphicsSettings.currentRenderPipeline?.GetType() ?? typeof(RenderPipelineAsset);
```

Comment:
you can use `RenderPipelineManager.currentPipeline` if you want the RenderPipeline type (our APIs are very confusing I know).
The problem I can see here is that this can be called from a InitializeOnLoad where we cannot guarantee that the ADB is ready - which means we cannot guarantee accessibility to the SRP asset of the current Quality level.
So I guess your solution works in some circumstances but not all. We definitely need to push for a callback to guarantee those scenarios. (OnAssetDatabaseReady?)